### DISCUSSION TITLE
cannot solve ModuleNotFoundError

### DISCUSSION THREAD

**Discussion by catdogmaus** [custom-components/pyscript#607] [Q&A]
I have a py file in use, the creator of which has lost interest in updating it. However, at the moment the only problem seems to be an error message

Exception in </config/pyscript/hewalex2mqtt.py> line 5: from hewalex_geco.devices import PCWU, ZPS ^ ModuleNotFoundError: No module named 'hewalex_geco'

the error is caused by the following line
`from hewalex_geco.devices import PCWU, ZPS`
where hewalex_geco.devices are folders where PCWU, ZPS files are located.
Is this bug? As I understand it should work!
Also when I use 
`from .hewalex_geco.devices import PCWU, ZPS`
I get: 
ImportError: attempted relative import with no known parent package
maybe someone has an idea how to fix this.

**Comment by stevius10** [comment]
Does hewalex2mqtt use PyScript or is it a regular Python script?
Based on 'from hewalex_geco.devices import PCWU, ZPS' I would expect PyScript to look for in modules/devices/. 

What I find on GitHub about hewalex2mqtt looks more like native Python or approaches that only use MQTT from Home Assistant or actually the broker itself. 
If this is the case and you still want to use it, for example in the context of events, you would first have to include the path in order to be able to execute Python natively: 

```python
  if PATH_DIR_PY_NATIVE not in sys.path:
    sys.path.append(PATH_DIR_PY_NATIVE)
```

Presumably there would also be possibilities to connect @pyscript_compile with dynamic imports, although this could quickly become confusing. But the question would be, if native, why the way via Pyscript.

**Reply by catdogmaus** [reply]
> But the question would be, if native, why the way via Pyscript.

I am not python programmer, so it is simply attempt to go on the road of least of resistance. As I understand pyscript should be trouble free way to use py scripts in HA. The second best is probably appdaemon but that requires already rewriting the script. 
Did I mention that I am not python programmer? 
Script that I tried to use is https://github.com/Chibald/Hewalex2Mqtt . As I understand it is not strictly meant for HA but when it would run as py script it should not matter. When output is mqtt, HA would be able to read it. Anyway, it seems that I am not able to use it. I did try out 

```
# Define the directory containing the hewalex_geco module
PATH_DIR_PY_NATIVE = "/homeassistant/pyscript"

# Ensure the directory is in sys.path
if PATH_DIR_PY_NATIVE not in sys.path:
    sys.path.append(PATH_DIR_PY_NATIVE)
```

but it seems to be unsuccessful. I stopped to have any errors about ModuleNotFoundError but also any other feedback from pyscript.

**Reply by stevius10** [reply]
PyScript is more like a sandbox environment, target to catch events occuring in your Home Assistant system. Actually one step ahead than "run as py script should not matter" - it does, especially when it comes to dependencies and runtime. 

The easiest way to integrate a Python script in Home Assistant is via shell_command or command_line sensor of Home Assistant. You can use the Python interpreter like you would on CLI. So both should accept a terminal command like "python3 ..". Could you start your script by CLI itself? So you can the same manner. 
This most likely doesn't solve dependencies itself, but Pyscript as an additional layer in a smaller sandbox environment can make to get your script running much more complicated.
